hw/chisel/doc/xdma.md

```diff
@@ -0,0 +1,238 @@
+
+## Source Code Hierarchy
+
+Root directory: ***snax_cluster/hw/chisel/src/main/scala/snax
+/xdma/***
+
+> - **CommonCells**
+>   - **ComplexQueue.scala**: A FIFO supporting multiple input, single output or single input, multiple output configurations with internal concatenation.
+>   - **CustomOperators.scala**: The custom operator to support the concatenation of UInt and insertion of pipeline registers.
+>   - **MuxDemuxDecoupled.scala**: Mux and Demux for decoupled signals with Valid/Ready handshaking.
+> - **DesignParams**
+>   - **DesignParams.scala**: Classes related to design parameters.
+> - **xdmaExtension**
+>   - **DMAExtension.scala**: Abstract base module for all DMA Extensions. All XDMA Extensions should extend from the base module declared in this file.
+>   - **MaxPool.scala**: Implementation of MaxPool Extension.
+>   - **Memset.scala**: Implementation of Memset Extension, allowing memory set operations other than zero. 
+>   - **Transposer.scala**: Implementation of matrix-transpose Extension for INT8 matrices. 
+> - **xdmaFrontend**
+>   - **DMACtrl.scala**: Controls the XDMA, managing configuration forwarding to or from remote sites.
+>   - **DMADataPath.scala**: Datapath of XDMA where instantiation of streamers and datapath extensions occur.
+> - **xdmaStreamer**
+>   - **AddressGenUnit.scala**: Address generation unit for all channels with configurable buffer sizes for generated addresses.
+>   - **DataRequestor.scala**: Module generating requests to the TCDM interface.
+>   - **DataResponser.scala**: Module receiving responses from the TCDM interface.
+>   - **Reader.scala**: Reads data from the TCDM, incorporating both a **DataRequestor** and a **DataResponser**.
+>   - **Writer.scala**: Writes data to the TCDM, incorporating a **DataRequestor**.
+> - **xdmaTop**
+>   - **xdmaTop.scala** The top module and entry point of the entire Scala (Chisel) program, acting as a SystemVerilog Generator.
+
+## Features
+
+This section introduces three key features of XDMA:
+
+### New Data-fetching Unit (Streamer) to Support Powerful Direct Memory Access
+
+The TCDM interconnect at the top of the figure provides full-region data access with a one-cycle delay if there is no contention. XDMA's configuration includes variable bit widths for both memory and AXI channels. The number of data streamers on both reader and writer sides is automatically determined by these configurations, e.g.,:
+
+- **TCDMDataWidth=64** and **AXIDataWidth=512**: Typical for Snax clusters. XDMA will instantiate eight channels for both reading and writing at the TCDM side, concatenating them into a single 512-bit channel at the AXI side.
+- **TCDMDataWidth=64** and **AXIDataWidth=64**: Typical for global scratchpad memory/SRAM. XDMA will instantiate only one channel on each side, adopting a simpler architecture used by most other DMAs.
```

Comment:
How does this look for L3 memory in the snax cluster? I thought this would be `**TCDMDataWidth=512** and **AXIDataWidth=512**`

Reply:
Yes. Another way is to use original idma as the "remote" xdma, which has strong affinity to L3 memory.